### DISCUSSION TITLE
Conflicting hotkey-mapping and hotkey-functions

### DISCUSSION THREAD

**Discussion by Stisen1** [ihhub/fheroes2#8137] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

When changing hotkeys for two separate types of key-functions, the game seem to either ignore key-inputs due to impossible simultanious command actions or choosing to only do one of the key-assigned commands. 

- For example: If you re-assign the hotkey for "kingdom summary" to "SPACE" instead of "K", and leave the hotkey for "Default action" as "SPACE", the game will ALWAYS prioritize opening the kingdom summary, while the "default action" become impossible to do!
- Furthermore: If you change the hotkey for "toggle fullscreen", "kingdom overview" AND "default action" to "SPACE", it will do the first two afformentioned actions simultaniously, but not do the default action.

### Describe the possible solution

Have a prompt pop-up with a notification, that speaks of conflicting key-mapping. It only need to say, that it might result in some command issues. Furthermore... Make it impossible to allow these types of conflicts. No one in their right mind would want one key to have more than one function at the same time. The same key for "default action" and "construction" renders no issues though, so there'll be no need for either prompting about this one or making it impossible to use this key for both these things.

**Comment by oleg-derevenetz** [comment]
> Make it impossible to allow these types of conflicts.

User can always edit the `fheroes2.key` manually, so such conflicts are always possible.

**Comment by Stisen1** [comment]
Sure, but if some commands can't happen simultaniously, and one of them overrule the other, it can be quite frustrating for mere mortals, who are used to getting a heads-up on impossible key-mappings. I know that in many games, you don't get that kind of luxury, but those games usually doesn't have like 50 cool hotkeys. And in this case, it seems as if you are FORCED to enter something as a hotkey, even if you don't want any :-)

**Comment by ihhub** [comment]
@Stisen1 , keys are divided into groups and even different events with the same key within the same group might be used in different places or dialogs. It is really hard to come up with logic about this unless we store some key-dependencies. Also as @oleg-derevenetz mentioned it is possible to modify a configuration file for hot keys which makes the job even harder.

**Comment by ihhub** [comment]
I am moving this topic to the discussion section as we have no solution for this problem.
